batchelor/__init__.py

```diff
@@ -349,106 +356,124 @@ def getListOfJobStates(self, jobIDs = None, username = None):
 			for section in self._config.sections():
 				if "memory" in [ e[0] for e in self._config.items(section)]:
 					self._config.set(section, "memory", memory)
 		if systemOverride == "local" and n_threads:
 			self._config.set("local","cores", n_threads);
-		
+
 		self._submittedJobs = []
 		self._commands = []
 		self._logfiles = []
 		self._check_job_success = check_job_success
 		self._store_commands = store_commands
 		self._store_commands_filename = ""
-		
+
 		if self._store_commands:
 			self._store_commands_filename = os.path.join(time.strftime("batchelorComandsLog_%y-%m-%d_%H-%M-%S.dat"))
 			if os.path.isdir(self._store_commands):
 				self._store_commands_filename = os.path.join(self._store_commands, self._store_commands_filename)
-			else: 
+			else:
 				self._store_commands_filename = os.path.join(os.getcwd(), self._store_commands_filename)
 
-	def submitJob(self, command, output = '/dev/null', wd = None, jobName=None, priority = None):
+	def submitJob(self, command, output = '/dev/null', wd = None, jobName=None, priority = None, waitFreeSlotsMax = None):
 		'''
 		Submit job with the given command
-		
+
 		@param command: Command to be executed
 		@param wd: Working directory. Default = current workingdirectory
-		@param output: Path or directory of log files. If not given, but check_job_success is selected, a .log folder will be created in the wd 
+		@param output: Path or directory of log files. If not given, but check_job_success is selected, a .log folder will be created in the wd
 		@param jobName: Name of the submitted job. Default='Batchelor'
-		
+		@param waitFreeSlotsMax: Number of seconds to wait maximal for a free slot if the system has a maxActiveJobs
+
 		@return: jobID
 		'''
 		if not jobName:
 			jobName = 'Batchelor'
 		if not wd:
 			wd = os.getcwd()
-			
+
 		if output == '/dev/null' and self._check_job_success: # if not output file is given but the job success should be checked anyway
 			logdir = os.path.join(wd, '.log')
 			if not os.path.isdir(logdir):
 				os.makedirs( logdir )
 			output = tempfile.mktemp(prefix = time.strftime("%Y-%m-%d_%H-%M-%S_"),suffix = '.log', dir = logdir)
-			
+
 		if self._check_job_success:
 			command = command + " && echo \"BatchelorStatus: OK\" || (s=$?; echo \"BatchelorStatus: ERROR ($s)\"; exit $s)"
-			
+
+		# check if only a certain amount of active jobs is allowd
+		if self._config.has_option(self._system, "maxActiveJobs"):
+			maxActiveJobs = int( self._config.get(self._system, "maxActiveJobs"))
+			i=0;
+			waitTime = 90
+			while True:
+				if i == 0:
+					sys.stdout.write("Waiting for free slots")
+					sys.stdout.flush()
+				if len(self.getListOfActiveJobs()) < maxActiveJobs:
+					break
+				time.sleep(waitTime); # wate 1.5  min
+				i+=1
+				if waitFreeSlotsMax != None and i*waitTime > waitFreeSlotsMax:
+					break
+					# raise BatchelorException("Could not  submit job because no free slot is available and the maximal wait time for a free slot is reached.")
```

Comment:
I guess there is a reason to test for the maximal number of active jobs, so why do you simply continue after the waiting time has passed? I would guess that for any batch system requiring this check, a following submission would also fail, finally raising an exception of the same type you commented here.